### PR TITLE
Adding Travis CI testing

### LINKED ISSUE #29: Add Travis testing
We should add a simple testing with Travis CI, at least for the official benchmark version.

### PR DESCRIPTION
**Pull request by tzanio**
Resolves #29

TODO
- [ ] Before merging, disable "Build Pushed Branches" in https://travis-ci.org/CEED/Laghos/settings
- [ ] Which version of MFEM to use (and cache): `laghos-1.0`, `master`, or `laghos-1.2`?
- [ ] Squash-merge

### CODE REVIEW

**Comment by tzanio** [comment]
Should be squash-merged

**Review by v-dobrev — APPROVED** [review]

**Comment by tzanio** [comment]
@vladotomov, any comments?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+     else
+        echo "Reusing cached metis-4.0/";
+     fi;
+
+   # MFEM
+   - if [ ! -e mfem/libmfem.a ]; then
+        rm -rf mfem;
+        git clone --depth 1 https://github.com/mfem/mfem.git;
+        cd mfem;
+        git checkout laghos-1.0;
```
Comment (vladotomov) [inline .travis.yml]: Should we make a new mfem tag?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
+language: cpp
+
+matrix:
+  include:
+    - os: linux
```
Comment (vladotomov) [inline .travis.yml:5]: Can we also test on macOS?

File: .travis.yml (hunk 1 above)
Comment (tzanio) [reply .travis.yml]: We will create an MFEM tag for the laghos-2.0 release, but I'm actually leaning towards testing MFEM master in the Laghosn Travis (we can make Laghis changes in the future that require changes in MFEM).

What do you think?

File: .travis.yml (hunk 2 above)
Comment (tzanio) [reply .travis.yml:5]: Yes, but I was thinking of adding that in a follow-on PR.

File: .travis.yml (hunk 1 above)
Comment (vladotomov) [reply .travis.yml]: I prefer master too.

**Review by vladotomov — APPROVED** [review]

**Pull request merged**